lib/browser/guest-window-manager.ts

```diff
@@ -59,6 +60,23 @@ export function openGuestWindow ({ embedder, guest, referrer, disposition, postD
     }
   }
 
+  if (createWindow) {
+    const webContents = createWindow({
+      webContents: guest,
+      ...browserWindowOptions
+    });
```

Comment:
Do we want to maybe throw an error here if `guest` is non-null and `webContents !== guest`?

i.e. a WebContents was passed in, but `createWindow` chose to return a different WebContents. I think that might cause a memory leak or other undesirable behavior, as the `guest` WebContents would still be hanging around. Especially, the browser may still navigate the `guest` WebContents even if it isn't installed into a window from `createWindow`.

e.g.
```js
const w = new BrowserWindow
w.webContents.setWindowOpenHandler((details) => ({
  action: 'allow',
  createWindow(opts) {
    const childWindow = new BrowserWindow() // no opts
    return childWindow.webContents // this is a fresh WebContents
  }
}))
await w.loadURL(`data:text/html,<script>
  const child = window.open('about:blank')
  child.document.documentElement.innerHTML = '<h1>hello</h1>'
</script>`)
```

In the above scenario, I think the behavior of this PR currently is that a blank child window would be opened, but because `guest` wasn't used when creating the child window, it would not be properly linked to the parent window. However, the child WebContents would still exist, it just wouldn't be displayed anywhere.

I think that's pretty surprising behavior and we should probably help developers avoid that situation. I'm not sure there's a really good argument to be made that it should be legal to return a WebContents that's different to the `{ webContents }` that's passed to `createWindow`, if there is one—if you want to cancel the window.open and create an unconnected WebContents, you can return `action: 'deny'` and create the unconnected child window yourself.